### PR TITLE
4.1 API

### PR DESCRIPTION
**Pull request by RachelTucker**
Generated 4.1.1 API
Added parsing of blob checksum headers for Head Object response

### CODE REVIEW

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java [hunk 1]
```diff
@@ -145,4 +149,90 @@ private static String readResponseString(final WebResponse response) throws IOEx
...
+     * Retrieves a map of blob offset to blob checksum from response headers.
+     */
+    public static ImmutableMap<Long, String> getBlobChecksumMap(final Headers headers) {
+        if (headers.keys() == null) {
+            return ImmutableMap.of();
+        }
+
+        final ImmutableMap.Builder<Long, String> builder = ImmutableMap.builder();
+
+        for (final String key : headers.keys()) {
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java]: You could update this to:

```java
headers.keys()
    .stream()
    .filter(key -> key.startsWith(BLOB_CHECKSUM_HEADER))
    .foreach(key -> builder.put(getOffsetFromHeaderKey(key), getBlobChecksumValue(key, headers)));
```

Comment (RachelTucker) [reply ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java]: are we using Java 8 in the SDK now?

Comment (rpmoore) [reply ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java]: Yes.

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java [hunk 2]
```diff
@@ -145,4 +149,87 @@ private static String readResponseString(final WebResponse response) throws IOEx
...
+    /**
+     * Converts the response header value for {@code BLOB_CHECKSUM_TYPE_HEADER} into a {@code ChecksumType.Type}
+     */
+    static ChecksumType.Type toBlobChecksumType(final String checksumType) {
+        if (Guard.isStringNullOrEmpty(checksumType)) {
+            return ChecksumType.Type.NONE;
+        }
+        switch (checksumType) {
+            case "CRC_32":
+                return ChecksumType.Type.CRC_32;
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java]: You should be able to use the built in `ChecksumType.Type.valueOf(checksumType)` method to do this same logic.

**Pull request merged**